src/modbus-tcp.c

```diff
@@ -208,6 +218,16 @@ static int _modbus_tcp_pre_check_confirmation(modbus_t *ctx, const uint8_t *req,
         return -1;
     }
 
+    /* Check unit ID */
+    if (rsp[6] != req[6]) {
```

Comment:
the others look ok, but this is risky, you will sometimes see broadcast addresses, then specific replies, I don't really like this one.

Reply:
@karlp Do you have some suggestion what to do instead?

Reply:
Just don't do this?  Work out what sort of buggy software you have that is causing the mixed data you were seeing.

Reply:
At one stage, @stephane  was proposing various "quirks" taht could be enabled/disabled, that might be useful?